src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java

```diff
@@ -0,0 +1,81 @@
+package com.wikia.webdriver.testcases.discussions;
+
+import com.wikia.webdriver.common.contentpatterns.MercurySubpages;
+import com.wikia.webdriver.common.contentpatterns.MercuryWikis;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.components.discussions.desktop.PostsCreatorDesktop;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.PostsCreatorMobile;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostsListPage;
+
+import org.testng.annotations.Test;
+
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_AUTO)
+public class CreatingPostTests extends NewTestTemplate {
+
+  private static final String DESKTOP_RESOLUTION = "1920x1080";
+  private static final String MOBILE_RESOLUTION = "600x800";
+
+  /**
+   * ANONS ON MOBILE SECTION
+   */
+
+  @Test(groups = "discussions-anonUserOnMobileCanSeePostDetailsList")
+  @Execute(asUser = User.ANONYMOUS)
+  @InBrowser(emulator = Emulator.GOOGLE_NEXUS_5)
+
+  public void anonUserOnMobileCanNotWriteNewPost() {
+    userOnMobileMustBeLoggedInToUsePostCreator();
+  }
+
+  /**
+   * ANONS ON DESKTOP SECTION
+   */
+
+  @Test(groups = "discussions-anonUserOnMobileCanSeePostDetailsList")
+  @Execute(asUser = User.ANONYMOUS)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+
+  public void anonUserOnDesktopCanNotWriteNewPost() {
+    userOnDesktopMustBeLoggedInToUsePostCreator();
+  }
+
+  /**
+   * TESTING METHODS SECTION
+   */
+
+  private void userOnMobileMustBeLoggedInToUsePostCreator() {
+    PostsCreatorMobile postsCreator = new PostsListPage().open().getPostsCreatorMobile();
+
+    Assertion.assertTrue(postsCreator.clickPostCreator().isModalDialogVisible());
+
+    postsCreator.clickOkButtonInSignInDialog();
+
+    Assertion.assertTrue(postsCreator.clickPostCreator().isModalDialogVisible());
+
+    postsCreator.clickSignInButtonInSignInDialog();
```

Comment:
@ludwikkazmierczak is there a way to handle situation when clicking the same button returns:
JoinPage for registered user
RegisterPage for anon user